### DISCUSSION TITLE
Cannot connect to DB using beforeAll() in Vitest VS Code Extension v0.6.0 but works in v0.4.1

### DISCUSSION THREAD

**Discussion by PhilippSalvisberg** [vitest-dev/vscode#345] [Q&A] [answered]
In this [repo](https://github.com/PhilippSalvisberg/js23c) I run some tests in the [`sandbox2`](https://github.com/PhilippSalvisberg/js23c/tree/main/sandbox2) folder with this `vite.config.ts`:

```typescript
import { defineConfig} from 'vitest/config'

export default defineConfig ({
  test: {
    pool: 'forks',
    poolOptions: {
      forks: {
        singleFork: true,
      },
      threads: {
        singleThread: true
      }
    }
  }
})

```

This works with `npm run test`, `npm run coverage` and it works also via the VS Code Extension 0.4.1. However, when I update the VS Code extension to 0.6.0 I get errors. The [dependencies](https://github.com/PhilippSalvisberg/js23c/blob/main/sandbox2/package.json) are correct IMO (vitest 1.5.0). 

The code executed in [beforeAll()](https://github.com/PhilippSalvisberg/js23c/blob/main/sandbox2/test/demotab.test.ts#L10-L11) fails with OracleDB related errors:

- NJS-512: invalid connection string parameters.
- NJS-514: syntax error in connection string

However, when I move the code from `beforeAll()` to a test then everything works as expected.

I'm running out of ideas. Is there some configuration that needs to be amended when running single-threaded tests via the VS Code extension 0.6.0? Or is this behaviour a bug?

Any help is very much appreciated.

**Comment by sheremet-va** [comment]
It should work the same way it does in a CLI. Unless you have some kind of configuration in a folder? Did you debug what values are passed down?

**Comment by PhilippSalvisberg** [comment]
>It should work the same way it does in a CLI. 

Good to know.

>Unless you have some kind of configuration in a folder? 

I duplicated the configuration for the CLI in the `vite.config.ts` mentioned above. For the CLI it's the following in `package.json`:

```json
        "test": "vitest --pool=forks --poolOptions.forks.singleFork --reporter=verbose --dir ./test",
```

I tried with `threads` and `forks`. Same result. 

>Did you debug what values are passed down?

Yes looks good. It's quite a simple configuration.

**Comment by ffMathy** [comment]
I had this same issue. I just moved all the "beforeAll" outside of "beforeAll". So in my setup files, I just moved that content in there directly.

For instance:

```
//setup.ts
beforeAll(...)
```

After:

```
//setup.ts
...
```

**Comment by PhilippSalvisberg** [comment]
@ffMathy thanks for the confirmation. Yes, as mentioned my code also works if I run it as part of the tests, outside of `beforeAll`.

So the workaround is to avoid `beforeAll` when using the Vitest VS Code extension? - Sounds like a bug to me. Especially, since it works as expected with the old version 0.4.1 of the extension.

**Reply by ffMathy** [reply, marked as answer]
Yeah I forgot to report this one during the development of the extension. I actually did detect it, but then due to the workaround, I forgot about it.

It's definitely a bug compared to how it was working before.

**Comment by PhilippSalvisberg** [comment]
I tried the new extension version v0.8.4. It works now as expected. Thanks a lot!
